src/frontend/query_commands.ml

```diff
@@ -502,6 +502,47 @@ let dispatch pipeline (type a) : a Query_protocol.t -> a =
     if path = "" then `Invalid_context else
       Locate.get_doc ~config
         ~env ~local_defs ~comments ~pos (`User_input path)
+  | Syntax_document (patho, pos) ->
+    let typer = Mpipeline.typer_result pipeline in
+    let pos = Mpipeline.get_lexing_pos pipeline pos in
+    let path =
```

Comment:
You don't use that at all, except for checking if it's empty. We probably don't need it. The only interesting thing is the node under the cursor.